File: 2 number sums/O(n)_time_and_space.js
function twoNumberSum(array, targetSum) {
  // Write your code here.
  let nums = {};
  for (let i of array) {
    if (targetSum - i in nums) {
      return [i, targetSum - i];
    } else {
      nums[i] = true;
    }
  }
  return [];
}

File: Stacks/Balanced Brackets/O(n)_time_and_space.py
def balancedBrackets(string):
    # Write your code here.
    opening="{[("
    closing="}])"
    matches={
        "}":"{",
        "]":"[",
        ")":"("
    }
    arr=[]

    for char in string:
        if char in opening:
            arr.append(char)
        elif char in closing:
            if len(arr) == 0:
                return False
            elif arr[-1] != matches[char]:
                return False
            else:
                arr.pop()
    return len(arr) == 0
